integrated_channels/integrated_channel/models.py

```diff
@@ -65,13 +67,27 @@ class Meta:
         verbose_name_plural = _("Enterprise Customer Plugin Configurations")
         unique_together = (("enterprise_customer_uuid", "enterprise_integrated_channel"),)
 
+    @property
+    def enterprise_customer(self):
+        """
+        Return Enterprise Customer associated with this instance.
+
+        Return :class:`enterprise.models.EnterpriseCustomer` instance associated with this
+        :class:`EnterpriseCustomerPluginConfiguration` instance via uuid.
+        """
+        try:
+            return EnterpriseCustomer.objects.get(uuid=self.enterprise_customer_uuid)
+        except EnterpriseCustomer.DoesNotExist:
+            return None
+
     def __str__(self):
         """
         Return human-readable string representation.
         """
-        return "<EnterpriseCustomerPluginConfiguration for enterprise {enterprise} using channel {channel}>".format(
-            enterprise=self.enterprise_customer_uuid,
-            channel=self.enterprise_integrated_channel.name
+        return "<EnterpriseCustomerPluginConfiguration for enterprise {} using channel {} for data type {}>".format(
+            (self.enterprise_customer.name if self.enterprise_customer else self.enterprise_customer_uuid),
```

Comment:
Yeah, seeing this here makes me think that we should introduce a "name" property for EnterpriseCustomer -- it's a weird case that we're describing here, however -- basically we have a UUID but not an actual record to go along with it -- if this is the case, where did the UUID even come from?